### PR TITLE
Unit tests on flow emulator

### PR DESCRIPTION
**Pull request by JawwadNFTGenius**
unit-tests-on-flow-emulator draft PR

### CODE REVIEW

File: unit-tests/contracts/Gaia.cdc [hunk 1]
```diff
@@ -0,0 +1,846 @@
+import NonFungibleToken from "./NonFungibleToken.cdc"
```
Comment (jctosta) [inline unit-tests/contracts/Gaia.cdc:1]: Why did you duplicated the contract files to your directory, the contracts should be the same for Flow Tokens or Dapper Utility Coins. Remove the contracts from your local directory and use the existing ones.

File: unit-tests/tests/03_txn_setup_account.test_ex.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    const nftAddrMap = { NonFungibleToken: serviceAccount }
+    
+    await deployContractByName("FungibleToken");
+    const ftAddrMap = { FungibleToken: serviceAccount }
+
+    await deployContractByName({name: "NFTStorefront", addressMap: nftAddrMap, ftAddrMap});
+    await deployContractByName({name: "DapperUtilityCoin", addressMap: ftAddrMap});
+
+    await shallPass(async () => {
+      return sendTransaction({ name: "setup_account" });
```
Comment (jctosta) [inline unit-tests/tests/03_txn_setup_account.test_ex.js:42]: @JawwadNFTGenius When you execute a transaction the blockchain state is changed in some way. Your test **should** validate that the intended change has actually been applied. 

The setup_account transaction creates the public collections under the user account, you must first ensure that the collections did not exist before the transaction was executed and then verify that they were created correctly.

File: unit-tests/tests/04_txn_create_set.test.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    //Deploying contracts used in create_templates
+    const serviceAccount = await getServiceAddress();
+    await deployContractByName("NonFungibleToken");
+    const nftAddrMap = { NonFungibleToken: serviceAccount }
+    
+    await deployContractByName({name: "Gaia", addressMap: nftAddrMap});
+    
+    await shallPass(async () => {
+      const args = ["Test Set", "Set Description", "https://www.ballerz.xyz/", "image uri", serviceAccount, 0.1];
+      return sendTransaction({ name: "create_set", args: args });
```
Comment (jctosta) [inline unit-tests/tests/04_txn_create_set.test.js:37]: @JawwadNFTGenius When you execute a transaction the blockchain state is changed in some way. Your test should validate that the intended change has actually been applied.

Check if the new Set has been created correctly, the contract has methods to validate this information.

File: unit-tests/contracts/Gaia.cdc (hunk 1 above)
Comment (JawwadNFTGenius) [reply unit-tests/contracts/Gaia.cdc]: Done

**Pull request closed without merging**